api/storage/client.go

```diff
@@ -90,6 +90,20 @@ func (c *Client) CreatePool(pname, provider string, attrs map[string]interface{}
 	return c.facade.FacadeCall("CreatePool", args, nil)
 }
 
+// DeletePool deletes the named pool
+func (c *Client) DeletePool(pname string) error {
```

Comment:
To be safe, should do api version check here too. Doesn't necessarily need another explicit test